From c247745faaf885fd3fa094198fc0ea288e295dbf Mon Sep 17 00:00:00 2001
From: Thibault Saunier <tsaunier@igalia.com>
Date: Fri, 13 Jul 2018 14:42:28 -0400
Subject: opencv: Fix build for opencv >= 3.4.2
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The `CV_RGB` macro is now in `imgproc.hpp`.

Fixes:

    ../subprojects/gst-plugins-bad/ext/opencv/gsthanddetect.cpp:497:40: error: ‘CV_RGB’ was not declared in this scope
            cvCircle (img, center, radius, CV_RGB (0, 0, 200), 1, 8, 0);
                                        ^~~~~~
---
 ext/opencv/MotionCells.cpp      | 3 +++
 ext/opencv/gsthanddetect.cpp    | 3 +++
 ext/opencv/gsttemplatematch.cpp | 3 +++
 3 files changed, 9 insertions(+)

diff --git a/ext/opencv/MotionCells.cpp b/ext/opencv/MotionCells.cpp
index f85989e..175ec90 100644
--- a/ext/opencv/MotionCells.cpp
+++ b/ext/opencv/MotionCells.cpp
@@ -51,6 +51,9 @@
 
 #include <errno.h>
 #include "MotionCells.h"
+#if (CV_MAJOR_VERSION >= 3)
+#include <opencv2/imgproc.hpp>
+#endif
 #include <opencv2/imgproc/imgproc_c.h>
 
 MotionCells::MotionCells ()
diff --git a/ext/opencv/gsthanddetect.cpp b/ext/opencv/gsthanddetect.cpp
index 60fd5be..47203fd 100644
--- a/ext/opencv/gsthanddetect.cpp
+++ b/ext/opencv/gsthanddetect.cpp
@@ -62,6 +62,9 @@
 
 /* element header */
 #include "gsthanddetect.h"
+#if (CV_MAJOR_VERSION >= 3)
+#include <opencv2/imgproc.hpp>
+#endif
 #include <opencv2/imgproc/imgproc_c.h>
 
 GST_DEBUG_CATEGORY_STATIC (gst_handdetect_debug);
diff --git a/ext/opencv/gsttemplatematch.cpp b/ext/opencv/gsttemplatematch.cpp
index f39208d..ec0b56a 100644
--- a/ext/opencv/gsttemplatematch.cpp
+++ b/ext/opencv/gsttemplatematch.cpp
@@ -63,6 +63,9 @@
 
 #include "../../gst-libs/gst/gst-i18n-plugin.h"
 #include "gsttemplatematch.h"
+#if (CV_MAJOR_VERSION >= 3)
+#include <opencv2/imgproc.hpp>
+#endif
 #include <opencv2/imgproc/imgproc_c.h>
 
 GST_DEBUG_CATEGORY_STATIC (gst_template_match_debug);
-- 
cgit v1.1
